Do not merge. Add support for managing battleships in Fleet.

iwle opened the pull request:
Summary of changes:

* Fix `toString()` bug in `Battleship`.
* Update `Fleet` to manage battleships and be able to deploy them on the map.
* Update `Fleet` to include new types of battleship.
* Update `PutShipCommand` to work together with updated `Fleet`.
* Update tests for `Battleship`, `Enemy`, `Player` and `Fleet`.

See #83.

iwle committed:
Fix toString() bug in  Battleship

Fix toString() bug in Battleship caused by redundant print statement.
Results in two names being printed where one is sufficient.

iwle committed:
Update tests for Battleship

Add tests for hashing, toString method, and default constructors.

iwle committed:
Update Fleet to include types of battleship

Update Fleet to include default number of battleships. Add methods that
deal with deployment of battleships onto map. Update tests for Player
and Enemy with new constructors.

iwle committed:
Add test for Enemy

Add test for Enemy to deal with new Fleet.

iwle committed:
Update PutShipCommand

Update PutShipCommand to work with Fleet. Add comments to Fleet. Fix
checkstyle errors.

erjillsison commented:
I don't think Fleet should be inside Parser because Parser should only handle parsing user input to create the Command. Can possibly put it in ModelManager first since Player has yet to be integrated.

Next, the counter for each deployed ship is updated regardless if PutShipCommand is successful or not.
Meaning that even a failed PutShipCommand will count towards the number of ships that you have deployed.

iwle commented:
Thanks for the review. In that case I will move the Fleet instance from Parser to ModelManager.

As for the other bug, I will throw the exception first if the Battleship cannot be deployed so that it will not decrement the count before a Battleship is deployed.

iwle committed:
Move fleet to ModelManager

Move fleet to ModelManager from PutShipCommandParser. Fix bug where a
ship that is not deployed is put on the map by checking if a ship can be
deployed first.

iwle committed:
Update tests for PutShipCommand
